src/Application/Game.cpp

```diff
@@ -1541,24 +1546,41 @@ void Game::EventLoop() {
                         LSTR_TOWN_PORTAL_TO_S, v69.c_str()));
                     continue;
                 }
-                case UIMSG_ShowFinalWindow: {
-                    // static due to GUIWindow_Modal not
-                    // holding a reference and text ptr will
-                    // be destroyed upon exiting scope
-                    static std::string final_message;
-
-                    final_message = StringPrintf(
-                        "%s\n \n%s\n \n%s",
-                        localization->GetString(LSTR_CONGRATULATIONS_ADVENTURER),
-                        localization->GetString(LSTR_WE_HOPE_YOU_ENJOYED_MM7),
-                        localization->GetString(LSTR_THE_MM7_DEV_TEAM));
-
-                    pModalWindow = new GUIWindow_Modal(final_message.c_str(), UIMSG_OnFinalWindowClose);
+                case UIMSG_ShowGameOverWindow: {
+                    pGameOverWindow = new GUIWindow_GameOver();
                     uGameState = GAME_STATE_FINAL_WINDOW;
                     continue;
                 }
-                case UIMSG_OnFinalWindowClose:
+                case UIMSG_OnGameOverWindowClose:
+                    pAudioPlayer->PauseSounds(-1);
+                    SaveGame(1, 0);
+
+                    pParty->vPosition.x = -17331;  // respawn in harmondale
```

Comment:
Just out of curiosity, where did you get this code from?

Reply:
Half pinched from GAME_STATE_PARTY_DIED